exercises/Stephen_Smith/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,119 @@
+#include "../include/ShapeSorter.h"
+
+
+
+// *************************************************//
+//Shape sorter
+//Constructors
+ShapeSorter::ShapeSorter()
+{
+
+};
+// Constructor with input list of shapes
+ShapeSorter::ShapeSorter(const std::vector<std::shared_ptr<BaseShape>>& shapeList)
```

Comment:
In C++11 with move semantics we could do this as

```c++
ShapeSorter::ShapeSorter(std::vector<std::shared_ptr<BaseShape>> shapeList) : m_shapeList(std::move(shapeList)) {}
```
to avoid having to copy the shapes in.